sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_utils.py

```diff
@@ -89,6 +89,26 @@ async def extract_data_template(feed_class, convert, feed_element):
     return next_link, iter(list_of_qd)  # when next_page is None, AsyncPagedItem will stop fetch next page data.
 
 
+async def extract_rule_data_template(feed_class, convert, feed_element):
+    """Special version of function extrat_data_template for Rule.
+
+    Pass both the XML entry element and the rule instance to function `convert`. Rule needs to extract
+    KeyValue from XML Element and set to Rule model instance manually. The msrest serialization/deserialization
+    doesn't work for this special part.
+    """
+    deserialized = feed_class.deserialize(feed_element)
+    next_link = None
+    if deserialized.link and len(deserialized.link) == 2:
+        next_link = deserialized.link[1].href
+    if deserialized.entry:
+        list_of_entities = [
+            convert(*x) if convert else x for x in zip(feed_element.findall(constants.ENTRY_TAG), deserialized.entry)
```

Comment:
since all that's different is whether you zip the ENTRY_TAG, would this not be better off as a flag onto a single "extract_data_template" method? Or is there more different that I'm not realizing.

Reply:
A single `extract_data_template` would work with an "if" statement. Both are acceptable to me.
I chose to put "Rule" related special code separately. In the near future with the autorest extension enhancement we should be able to remove this special `extract_rule_data_template` without changing `extract_data_template`. Let me know if you think a single method is better. I'm open to either choice.

Reply:
++ thanks for the context.  Normally I'd say yes, they'd be better off as one, but your point about deprecation is very well taken.  Let's add a note to the comment explaining this plan (because I will inevitably not remember a month from now) and I can live with this for the time being.

Reply:
Added docstring to "extract_rule_data_template" to explain it.